src/lib.rs

```diff
@@ -1,20 +1,75 @@
 #![no_std]
-/// MQTT Request/response Handling
-///
-/// # Overview
-/// This library is intended to be an easy way to handle inbound requests automatically.
-///
-/// Handler functions can be associated with the library to be automatically called whenever a
-/// specified request is received, and the handler will automatically be invoked when the request
-/// is received.
-///
-/// ## Limitations
-/// * The `poll()` function has a somewhat odd signature (using a function to provide the `Context`
-/// and call the handler) due to required compatibility with RTIC and unlocked resources.
-///
-/// * Handlers may only be closures that do not capture any local resources. Instead, move local
-/// captures into the `Context`, which will be provided to the handler in the function call.
-///
+//! MQTT Request/response Handling
+//!
+//! # Overview
+//! This library is intended to be an easy way to handle inbound requests automatically.
+//!
+//! Handler functions can be associated with the library to be automatically called whenever a
+//! specified request is received, and the handler will automatically be invoked with the request
+//! data.
+//!
+//! ## Limitations
+//! * The `poll()` function has a somewhat odd signature (using a function to provide the `Context`
+//! and call the handler) due to required compatibility with RTIC and unlocked resources.
+//!
+//! * Handlers may only be closures that do not capture any local resources. Instead, move local
+//! captures into the `Context`, which will be provided to the handler in the function call.
+//!
+//! ## Example
```

Comment:
Is there a way to test this in CI? As a doctest or as an `include!()` maybe?

Reply:
Example code is already tested via CI for compile (`cargo test` verifies this runs properly). I've intentionally prefixed it as `no_run` because it has a `loop {}` that never exits, so the `cargo test` command would hang otherwise

Reply:
Ack. But is there a way to ensure that this example in the documentation does not diverge from the other examples/the API?

Reply:
The only way I think we can guarantee that is by making this equivalent to the existing `integration_test.rs`, but I feel that would be pretty overkill and difficult to accomplish.

As it stands, we are forced to update this already if any of the currently-used API changes, so this would only become invalid if we made some fundamental changes to the design. I'd guess that's somewhat unlikely to occur, but I see where you're coming from.

Personally, I think it may be more trouble that it's worth to ensure this is a fully-functional example.